### PR TITLE
Collaboration API schema

### PR DESCRIPTION
**Pull request by echappen**
## What’s changing?

Adds internal API schema for collaboration on match resolution representing two endpoints:

1. Get Match Details 
2. Update Match Disposition

## Why?

This data will feed the UI in the Collaboration Tool for state collaboration.

Jira/NAC-131

## This PR has:

- [ ] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)
- [X] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)
- [ ] **Automated unit tests** (_to maintain or increase level of code coverage_)
- [ ] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)

## Anything else?

Since this API relies so heavily on Match data, I kept it in the `match` subsystem instead of breaking it out into a `collaboration` subsystem like we were initially thinking. 

I'm also reconsidering naming this the "Collaboration" API and instead calling it the "Match Resolution API". The latter is more descriptive and I find myself reaching for it more often for descriptions in this API schema.

I still need to find a way to represent original query and response data from the match.

### CODE REVIEW

File: match/docs/openapi/resolution/disposition_update.yaml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                initial_action_at:
+                  type: string
+                  format: date
+                  description: "The timestamp of the initial action taken by the state"
+                invalid_match:
+                  type: boolean
+                  description: "The state's determination of whether the match is invalid"
+                protect_location:
+                  type: boolean
+                  description: "Determines whether states should protect location of the individual when resolving matches"
```
Comment (kingcomma) [inline match/docs/openapi/resolution/disposition_update.yaml:50]: *Thought (non-blocking):* Something to chew on as we get further into this implementation: do we want these values to be nullable or have a default value? A `null` value may more accurately represent situations where a state has not yet made a determination for a given field.

File: match/docs/openapi/resolution/match_show.yaml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                properties:
+                  match_id:
+                    type: string
+                    description: "The Match ID"
+                    example: "EBOINR"
+                  status:
+                    type: string
+                    description: "The Match's status"
+                    example: "open"
+                  state_initial_data:
```
[the remark this answers is not in the document]
Comment (echappen) [reply match/docs/openapi/resolution/match_show.yaml:51]: Yep

File: match/docs/openapi/resolution/disposition_update.yaml [hunk 3]
```diff
@@ -0,0 +1,84 @@
+put:
```
[the remark this answers is not in the document]
Comment (echappen) [reply match/docs/openapi/resolution/disposition_update.yaml:1]: Yep, that makes sense

File: match/docs/openapi/resolution/disposition_update.yaml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+          InitialAction:
+            description: "Example of a state providing the date of their initial action taken"
+            value:
+              data:
+                initial_action_at: "2022-05-06"
+          InvalidMatch:
+            description: "Example of a state claiming the match is invalid"
+            value:
+              data:
+                invalid: true
```
Comment (kingcomma) [inline match/docs/openapi/resolution/disposition_update.yaml:61]: *Todo:* looks like there's a mismatch between `invalid` here and `invalid_match` above.

File: match/docs/openapi/resolution/match_show.yaml [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+            type: object
+            properties:
+              data:
+                type: object
+                description: "The response payload representing match data."
+                required:
+                  - match_id
+                  - status
+                  - state_initial_data
+                  - state_disposition_data
```
Comment (kingcomma) [inline match/docs/openapi/resolution/match_show.yaml:41]: *Nitpick:* Looks like these names need to be updated too.

File: match/docs/openapi/resolution/disposition_update.yaml [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+          StateParticipantInfo:
+            description: "Exmaple of a state providing the participant data in their system"
+            value:
+              data:
+                case_id: "123XYZ"
+                participant_id: "890ABC"
+                benefits_end_at: "2022-5-31"
+                recent_benefit_months: ["2022-4", "2022-3", "2022-2"]
+                protect_location: false
```
Comment (kingcomma) [inline match/docs/openapi/resolution/disposition_update.yaml:70]: *Question:* Missed this on my last review. Are you still intending for participant data to be updatable?

**Comment by echappen** [comment]
@edwintorres No review needed but tagging you here, Edwin so you can follow along for your FE work

**Comment by echappen** [comment]
@kcherukupalli Do you mind reviewing this sometime today?

**Review by kcherukupalli — APPROVED** [review]
Looks good.

**Comment by echappen** [comment]
@kcherukupalli Thank you!

**Pull request merged**